### ISSUE TITLE
Image runs as root

### ISSUE THREAD

**Issue by alex1989hu** [DependencyTrack/frontend#48]
### Current Behavior:
The currently used base image runs as root:
https://github.com/DependencyTrack/frontend/blob/53a1734b7438d336367f3a80a622e9d79cd5ab73/docker/Dockerfile#L22 

### Steps to Reproduce:
Check the `Dockerfile` reference above.

### Expected Behavior:
Use the official [docker-nginx-unprivileged](https://github.com/nginxinc/docker-nginx-unprivileged) as base image.

### Environment:
  
  - Dependency-Track Version: `1.1.0`
  - Client Browser: `N/A`
  - Client O/S: `N/A`

### Additional Details:
`N/A`

**Comment by stevespringett** [comment]
PR from #49 resulted in broken image

**Comment by alex1989hu** [comment]
> PR from #49 resulted in broken image

Not broken: https://github.com/DependencyTrack/frontend/pull/49#issuecomment-776453352

**Comment by davidkarlsen** [comment]
It would be good to have container-friendly images both for the front-end and backend, i.e. images that:
* can run as non-root (either a random uid, or a fixed one)
* run with readOnlyRootFs (i.e. container is immutable, but writes to specific volumes like /tmp, /data etc

we're not able to upgrade the helm-chart to run the 4.x series on Kubernetes before this is in place.

**Comment by stevespringett** [comment]
Actually it was broken. The container did not respond to HTTP requests. I would consider that broken. Now if that was an environmental issue on my side, it's possible, but I need to be able to build and test the containers and I was unable to successfully test the container because my browser was not able to communicate with nginx. 

Ideally, there would only be a single Dockerfile. I'm not interested in maintaining multiple.

**Comment by stevespringett** [comment]
Also keep in mind that whatever changes are made to the Dockerfile, need to be compatible with the docker-compose that we publish. This is how the majority of people start adopting DT and its the way I test DT. So if changes were made to the Dockerfile that were incompatible with the current compose file, that may explain why testing failed.

**Comment by alex1989hu** [comment]
> Also keep in mind that whatever changes are made to the Dockerfile, need to be compatible with the docker-compose that we publish. This is how the majority of people start adopting DT and its the way I test DT. So if changes were made to the Dockerfile that were incompatible with the current compose file, that may explain why testing failed.

In my PR I also updated the compose file.

**Comment by davidkarlsen** [comment]
@stevespringett Is anything of this covered by integration-tests - that could help a bit?

**Comment by stevespringett** [comment]
@davidkarlsen unfortunately, no integration tests at the moment. All unit tests take around 35-40 minutes to complete.  Ideally, once that completes successfully, integration tests would kick off. That's not in place today.

**Comment by stevespringett** [comment]
I'd like to get this resolved and simply publish an updated Docker image without having to wait for the next version of DT.

**Comment by davidkarlsen** [comment]
@stevespringett will you factor out the docker stuff into a separate repo?

**Comment by stevespringett** [comment]
@davidkarlsen I don't understand the question. Can you rephrase?

I have no intention on moving Docker build and image deployment to a different repo, if that's what you're asking.

**Comment by stevespringett** [comment]
I'd like to get this issue resolved in the next two weeks or so as DT 4.2 release is quickly approaching.

**Comment by davidkarlsen** [comment]
> @davidkarlsen I don't understand the question. Can you rephrase?
> 
> I have no intention on moving Docker build and image deployment to a different repo, if that's what you're asking.

Yes, that was what I was asking.

**Comment by davidkarlsen** [comment]
> I'd like to get this issue resolved in the next two weeks or so as DT 4.2 release is quickly approaching.

Great - so maybe a test with compose can be added to vet any container changes. I added a PR, but it was in a separate job, and the compose stuff seems to require a dist directory. Maybe you want to massage that a bit (either in a separate job and use containers, or in the build-step and use the "dist" directory).

Then we can come back with changes to meet container best-practices and have it verified in CI.

**Comment by stevespringett** [comment]
Closing as resolved. Thanks for the help everyone. Open a new ticket if additional changes are required.
